runtime/pkg/activity/kafka.go

```diff
@@ -53,12 +57,72 @@ func NewKafkaSink(brokers, topic string, logger *zap.Logger) (*KafkaSink, error)
 		return nil, err
 	}
 
-	return &KafkaSink{
+	sink := KafkaSink{
 		producer: producer,
 		topic:    topic,
 		logger:   logger,
 		logChan:  logChan,
-	}, nil
+	}
+
+	go sink.processProducerEvents(producer, logger)
+
+	return &sink, nil
+}
+
+func (s *KafkaSink) SetActivity(activity Client) {
+	s.activity = activity
+}
+
+func (s *KafkaSink) processProducerEvents(producer *kafka.Producer, logger *zap.Logger) {
+	var deliverySuccessCount int
+	var deliveryFailureCount int
+	var lastDeliveryError error
+
+	ticker := time.NewTicker(reportStatsPeriod)
+	defer ticker.Stop()
+
+	reportDeliveryStats := func() {
+		if deliverySuccessCount > 0 {
+			if s.activity != nil {
+				s.activity.Emit(context.Background(), "activity_kafka_delivery_success", float64(deliverySuccessCount))
+			}
+			deliverySuccessCount = 0
+		}
+		if deliveryFailureCount > 0 {
+			logger.Warn(fmt.Sprintf("Kafka sink: delivery failures in the last minute: %d", deliveryFailureCount),
+				zap.Error(lastDeliveryError))
+			if s.activity != nil {
+				// Be optimistic - if delivery failed because of a temporary issue then
+				// the following activity event might be eventually delivered
+				s.activity.Emit(context.Background(), "activity_kafka_delivery_failure", float64(deliveryFailureCount),
+					attribute.String("error", lastDeliveryError.Error()))
+			}
+			deliveryFailureCount = 0
+			lastDeliveryError = nil
+		}
+	}
```

Comment:
We should use OpenTelemetry for this. OpenTelemetry is still our primary choice for internal observability. I think we should aim to keep the activity client to the original use case of gathering events for usage, billing and uptime reporting.

That will also avoid the cyclic dependency between the sink and activity client, which is a nice benefit.

Eventually, we should probably find a way to emit OpenTelemetry data to the Rill metrics project as well.